### PR TITLE
#36897 Create new endpoint to store uploaded file in temporary storage

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/web/rest/TemporaryResourceStorageResource.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.multipart.MultipartFile
+
+@RestController
+@RequestMapping("/api")
+class TemporaryResourceStorageResource(
+    private val resourceService: TemporaryResourceStorageService
+) {
+
+    @PostMapping(value = ["/resource/temp"], consumes = [MULTIPART_FORM_DATA_VALUE])
```
Comment (theo-ritense) [inline resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/web/rest/TemporaryResourceStorageResource.kt:36]: Please move this path to the class-level @RequestMapping

File: resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/autoconfigure/TemporaryResourceStorageAutoConfiguration.kt [hunk 2]
```diff
@@ -44,4 +47,19 @@ class TemporaryResourceStorageAutoConfiguration {
         return TemporaryResourceStorageDeletionService(retentionInMinutes)
     }
 
+    @Bean
+    @ConditionalOnMissingBean(TemporaryResourceStorageResource::class)
+    fun temporaryResourceStorageResource(
```
Comment (theo-ritense) [inline resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/autoconfigure/TemporaryResourceStorageAutoConfiguration.kt:52]: We definitely need more 'resource' in this name.

File: resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/web/rest/TemporaryResourceStorageResource.kt (hunk 1 above)
Comment (Klaas-Ritense) [reply resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/web/rest/TemporaryResourceStorageResource.kt]: changed

File: resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/autoconfigure/TemporaryResourceStorageAutoConfiguration.kt (hunk 2 above)
Comment (Klaas-Ritense) [reply resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/autoconfigure/TemporaryResourceStorageAutoConfiguration.kt:52]: We also have a `ResourceResource` :p

**Review by theo-ritense — APPROVED** [review]

**Pull request merged**